<?xml version="1.0"?>
<robot name="reach5mini">

<material name="black">
  <color rgba="0 0 0 1"/>
</material>

  <link name="base_link">
    <visual>
      <geometry>
	<mesh filename="package://reach_system_one_support/meshes/reach5mini/visual/Base.stl" />
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.150" radius="0.020"/>
      </geometry>
      <origin rpy="0 1.57075 0" xyz="-0.075 0 0"/>
    </collision>
    <inertial>
      <mass value="0.341"/>
      <origin xyz="-0.075 -0.006 -0.003"/>
      <inertia ixx="0.000099" ixy="0.000139" ixz="0.000115" iyy="0.002920" iyz="0.000003" izz="0.002934"/>
    </inertial>
  </link>
  
  <link name="shoulder_link">
    <visual>
      <geometry>
        <mesh filename="package://reach_system_one_support/meshes/reach5mini/visual/ShoulderJoint.stl" />
      </geometry>
      <material name="black"/>
      <origin rpy="0 0 -1.57075" xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.066" radius="0.020"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.010"/>
    </collision>
    <inertial>
      <mass value="0.194"/>
      <origin xyz="0.005 -0.001 0.016"/>
      <inertia ixx="0.000189" ixy="0.000005" ixz="0.000054" iyy="0.000213" iyz="0.000003" izz="0.000067"/>
    </inertial>
  </link>
    
  <link name="upper_arm_link">
    <visual>
      <geometry>
        <mesh filename="package://reach_system_one_support/meshes/reach5mini/visual/UpperArm.stl" />
      </geometry>
      <origin rpy="0 -1.57075 -0.2655555" xyz="0 0 -0.075"/>
      <material name="black"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.1445" radius="0.020"/>
      </geometry>
      <origin rpy="0 1.57075 -0.27075" xyz="0.075 0 0"/>
    </collision>
    <collision>
      <geometry>
        <cylinder length="0.040" radius="0.020"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>    
    </collision>
    <inertial>
      <mass value="0.429"/>
      <origin xyz="0.073 0.0 0.0"/>
      <inertia ixx="0.000087" ixy="-0.000076" ixz="-0.000010" iyy="0.003190" iyz="0.000000" izz="0.003213"/>
    </inertial>
  </link>

  <link name="forearm_link">
    <visual>
      <geometry>
        <mesh filename="package://reach_system_one_support/meshes/reach5mini/visual/ForearmJoint.stl" />
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.040" radius="0.020"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <mass value="0.115"/>
      <origin xyz="0.017 -0.026 -0.002"/>
      <inertia ixx="0.000120" ixy="-0.000061" ixz="-0.000001" iyy="0.000062" iyz="0.000000" izz="0.000156"/>
    </inertial>
  </link>	

  <link name="wrist_link">
    <visual>
      <geometry>
        <mesh filename="package://reach_system_one_support/meshes/reach5mini/visual/Wrist.stl" />
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.140" radius="0.020"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.070"/>
    </collision>
    <inertial>
      <mass value="0.333"/>
      <origin xyz="0.0 0.003 -0.098"/>
      <inertia ixx="0.003709" ixy="-0.000002" ixz="-0.000004" iyy="0.003734" iyz="0.000000" izz="0.000079"/>
    </inertial>
  </link>

  <link name="tip">
  </link>
  
  <link name="jaw1">
    <visual>
      <geometry>
      	<mesh filename="package://reach_system_one_support/meshes/reach5mini/visual/Jaw1.stl" />
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.03 0.09 0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.01 -0.045 0"/>
    </collision>
  </link>

  <link name="jaw2">
    <visual>
      <geometry>
      	<mesh filename="package://reach_system_one_support/meshes/reach5mini/visual/Jaw2.stl" />
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black"/>      
    </visual>
    <collision>
      <geometry>
        <box size="0.03 0.09 0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.01 -0.045 0"/>
    </collision>
  </link>


  <joint name="joint1" type="revolute">
    <parent link="base_link"/>
    <child link="shoulder_link"/>
    <origin xyz="0 0 0" rpy="0 0 0"/> <!--Was <origin xyz="0 0 0" rpy="0 0 3.141592/> before-->
    <axis xyz="0 0 1"/>
    <limit effort="9.0" lower="0.0" upper = "6.2" velocity="0.5"/>
  </joint>

  <joint name="joint2" type="revolute">
    <parent link="shoulder_link"/>
    <child link="upper_arm_link"/>
    <origin xyz="0.020 0 0.046" rpy="1.57075 1.3 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="9.0" lower="0.0" upper = "3.5" velocity="0.5"/>
  </joint>

  <joint name="joint3" type="revolute">
    <parent link="upper_arm_link"/>
    <child link="forearm_link"/>
    <origin xyz="0.15 0 0" rpy="3.1415 0 1.3"/>
    <axis xyz="0 0 1"/>
    <limit effort="9.0" lower="0.0" upper = "3.5" velocity="0.5"/>
  </joint>

  <joint name="joint4" type="revolute">
    <parent link="forearm_link"/>
    <child link="wrist_link"/>
    <origin xyz="0.020 0 0" rpy="-1.57075 0 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="5.0" lower="0.0" upper = "6.0" velocity="1.0"/>
  </joint>

  <joint name="joint5" type="revolute">
    <parent link="wrist_link"/>
    <child link="jaw1"/>
    <origin xyz="0.015 0 -0.140" rpy="1.57075 0 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="5.0" lower="0.0" upper = "0.5" velocity="0.5"/>
  </joint>

  <joint name="joint6" type="revolute">
    <parent link="wrist_link"/>
    <child link="jaw2"/>
    <origin xyz="-0.015 0 -0.140" rpy="1.57075 0 0"/>
    <axis xyz="0 0 -1"/>
    <limit effort="5.0" lower="0.0" upper = "0.5" velocity="0.5"/>
  </joint>

  <joint name="arm_to_tip" type="fixed">
    <parent link="wrist_link"/>
    <child link="tip"/>
    <origin xyz="0 0 -0.180" rpy="0 1.57075 0"/>
  </joint>
  

</robot>
